docs/.vitepress/lang.js

```diff
@@ -1,24 +1,28 @@
 ;(() => {
   const supportedLangs = window.supportedLangs
-  let userPreferredLang = localStorage.getItem('preferred_lang')
-
-  if (!userPreferredLang) {
-    const systemLang = navigator.language
-    localStorage.setItem('preferred_lang', systemLang)
-    userPreferredLang = systemLang
+  const cacheKey = 'preferred_lang'
+  const defaultLang = 'en-US'
+  // docs supported languages
+  const langAlias = {
+    en: 'en-US',
+    fr: 'fr-FR',
+    es: 'es-ES',
   }
-
-  if (
-    supportedLangs.includes(userPreferredLang) &&
-    !location.pathname.startsWith(`/${userPreferredLang}`)
-  ) {
+  let userPreferredLang = localStorage.getItem(cacheKey) || navigator.language
```

Comment:
Maybe we can use `useLocalStorage`.
https://vueuse.org/core/uselocalstorage/#uselocalstorage

Reply:
This is pure JavaScript injection, running without `Vue` runtime, we cannot use that.